overlord/snapstate/refresh_control.go

```diff
@@ -0,0 +1,173 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package snapstate
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/mount"
+	"github.com/snapcore/snapd/overlord/ifacestate/ifacerepo"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+)
+
+var gateAutoRefreshHookName = "gate-auto-refresh"
+
+func affectedByRefresh(st *state.State, updates []*snap.Info) ([]string, error) {
```

Comment:
we really need also to keep track of which snaps  are affecting which, otherwise we don't know what updates to hold

Reply:
Done. I'm not sure the simple structure (map of maps) is the best long term, maybe we will need more rich info and actual structs, but it's not clear at the moment.

Reply:
Changed to a struct to convey info about base/restart.